packages/edit-site/src/components/dataviews/view-actions.js

```diff
@@ -199,38 +183,41 @@ function SortMenu( { fields, view, onChangeView } ) {
 					}
 				>
 					{ __( 'Sort by' ) }
-				</DropdownSubMenuTriggerV2>
+				</DropdownMenuItemV2Ariakit>
 			}
 		>
 			{ sortableFields?.map( ( field ) => {
 				const sortedDirection = view.sort?.direction;
 				return (
-					<DropdownSubMenuV2
+					<DropdownMenuV2Ariakit
 						key={ field.id }
 						trigger={
-							<DropdownSubMenuTriggerV2
+							<DropdownMenuItemV2Ariakit
 								suffix={ <Icon icon={ chevronRightSmall } /> }
 							>
 								{ field.header }
-							</DropdownSubMenuTriggerV2>
+							</DropdownMenuItemV2Ariakit>
 						}
-						side="left"
+						placement="left-start"
 					>
 						{ Object.entries( sortingItemsInfo ).map(
 							( [ direction, info ] ) => {
-								const isActive =
-									currentSortedField &&
+								const isChecked =
+									currentSortedField !== undefined &&
 									sortedDirection === direction &&
 									field.id === currentSortedField.id;
+
 								return (
-									<DropdownMenuItemV2
+									<DropdownMenuRadioItemV2Ariakit
 										key={ direction }
-										prefix={ <Icon icon={ info.icon } /> }
-										suffix={
-											isActive && <Icon icon={ check } />
-										}
-										onSelect={ ( event ) => {
-											event.preventDefault();
+										value={ direction }
+										name={ `view-actions-sorting-${ field.id }` }
+										suffix={ <Icon icon={ info.icon } /> }
+										// Note: there is currently a limitation from the DropdownMenu
+										// component where the radio won't unselect when all related
+										// radios are set to false.
+										checked={ isChecked }
```

Comment:
Just flagging that currently it looks like `DropdownMenuRadioItem` is not able to support unselecting a previously selected radio item. I'm currently looking into it.

Reply:
Ok, so I looked a bit more into this, specifically related to the sorting options. Here are some thoughts:

- using checkboxes (like currently on `trunk`) isn't a good idea, since users don't expect that selecting a checkbox will de-select another checkbox;
- using radio seems to be the right choice since we're talking about selecting mutually exclusive options
- however, radio buttons are not really supposed to be de-selectable once selected (see for example [this conversation](https://ux.stackexchange.com/questions/13511/why-is-it-impossible-to-deselect-html-radio-inputs))

I think that the cleanest solution, at least in terms of semantics, accessibility, and ultimately UX may be to add an extra option labelled something like "None" or "Default" (or anything else that conveys the notion to the user that selecting it will remove the ascending/descending sorting).

Reply:
Might want to get a design check here. @SaxonF @jameskoster

Reply:
Also noting that this implementation changes the design a bit by always keeping the `check` as a prefix. This affects the sorting submenu.

<img width="782" alt="Screenshot 2023-11-20 at 12 01 45 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/975ebaca-c74e-4ffb-b27b-615af480030f">

So that would also need some design input.

Reply:
We received [design input](https://github.com/WordPress/gutenberg/issues/55933) and I'm in the process of applying it to the component (https://github.com/WordPress/gutenberg/pull/56041). After that (and a couple of other PRs) are merged, I should be able to rebase and resume the work on this PR :)